build issues for gcc-canadian-cross

Happy New Year! 

Unfortunately the last changes make some trouble - again the problem is, that `/usr/include` is used.
No idea why the sysroot is ignored there. https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html
But i don't understand this part:

> Add the directory dir to the list of directories to be searched for header files during preprocessing. If dir begins with ‘=’ or $SYSROOT, then the ‘=’ or $SYSROOT is replaced by the sysroot prefix; see --sysroot and -isysroot. 

includes are never prefixed with `‘=’ or $SYSROOT`, at least i never saw this so far?! btw: it works if i add the `‘=’` manually in the build-shell. 🤷‍♂️

errorlog:
```
Making all in include
make[4]: Entering directory '/home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include'
mkdir -p ./x86_64-bob_compat-linux-gnu/bits/stdc++.h.gch
mkdir -p ./x86_64-bob_compat-linux-gnu/bits/stdc++.h.gch
x86_64-bob_compat-linux-gnu-c++  --sysroot=/home/mhaase/fdt.bobrecipes/dev/dist/devel/compat/cross-toolchain/1/workspace/usr/bin/../sysroots/x86_64-bob_isolation-linux-gnu   -x c++-header -nostdinc++ -g -O2 -D_GNU_SOURCE  -I/home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include/x86_64-bob_compat-linux-gnu -I/home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include -I/home/mhaase/fdt.bobrecipes/dev/src/devel/compat/gcc/1/workspace/libstdc++-v3/libsupc++ -I/usr/include  -O2 -g -std=gnu++0x /home/mhaase/fdt.bobrecipes/dev/src/devel/compat/gcc/1/workspace/libstdc++-v3/include/precompiled/stdc++.h \
-o x86_64-bob_compat-linux-gnu/bits/stdc++.h.gch/O2ggnu++0x.gch
x86_64-bob_compat-linux-gnu-c++  --sysroot=/home/mhaase/fdt.bobrecipes/dev/dist/devel/compat/cross-toolchain/1/workspace/usr/bin/../sysroots/x86_64-bob_isolation-linux-gnu   -x c++-header -nostdinc++ -g -O2 -D_GNU_SOURCE  -I/home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include/x86_64-bob_compat-linux-gnu -I/home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include -I/home/mhaase/fdt.bobrecipes/dev/src/devel/compat/gcc/1/workspace/libstdc++-v3/libsupc++ -I/usr/include  -O2 -g /home/mhaase/fdt.bobrecipes/dev/src/devel/compat/gcc/1/workspace/libstdc++-v3/include/precompiled/stdc++.h -o x86_64-bob_compat-linux-gnu/bits/stdc++.h.gch/O2g.gch
In file included from /usr/include/features.h:392:0,
                 from /home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include/x86_64-bob_compat-linux-gnu/bits/os_defines.h:39,
                 from /home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include/x86_64-bob_compat-linux-gnu/bits/c++config.h:533,
                 from /home/mhaase/fdt.bobrecipes/dev/build/devel/compat/gcc-canadian-cross/x86_64-bob_isolation-linux-gnu/1/workspace/build/x86_64-bob_compat-linux-gnu/libstdc++-v3/include/cassert:43,
                 from /home/mhaase/fdt.bobrecipes/dev/src/devel/compat/gcc/1/workspace/libstdc++-v3/include/precompiled/stdc++.h:33:
/usr/include/features-time64.h:21:10: fatal error: bits/timesize.h: No such file or directory
 #include <bits/timesize.h>
          ^~~~~~~~~~~~~~~~~
```

--- a/recipes/devel/compat/gcc.yaml
+++ b/recipes/devel/compat/gcc.yaml
@@ -22,7 +22,8 @@ checkoutDeterministic: True
 checkoutScript: |
     patchApplySeries -p1 \
         $<<gcc/0001-squash-multilib-libdir-suffix.patch>> \
-        $<<gcc/buildroot-libtool-v2.2.patch>>
+        $<<gcc/buildroot-libtool-v2.2.patch>> \
+        $<<gcc/0003-fix-canadian-cross-includes.patch>>
 
 buildTools: [host-toolchain]
 buildVars: [AUTOCONF_BUILD, AUTOCONF_HOST, AUTOCONF_TARGET,

--- a/recipes/devel/compat/gcc/0003-fix-canadian-cross-includes.patch
+++ b/recipes/devel/compat/gcc/0003-fix-canadian-cross-includes.patch
@@ -1,0 +1,25 @@
+Fix libstdc++ include paths for candian cross builds
+
+In a true canadian cross build, libstdc++ is built with the "target" compiler.
+This is a cross compiler and it is therefore harmful to use the build machine
+standard include paths. Instead, we can rely on the fact that the C and libc
+header files are found naturally by the cross-compiler in its sysroot.
+
+Upstream bug report: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71212
+
+Signed-off-by: Jan Klötzke <jan@kloetzke.net>
+
+--- a/libstdc++-v3/configure	2018-08-13 21:16:02.093382000 +0200
++++ b/libstdc++-v3/configure	2024-01-05 22:03:47.161321488 +0100
+@@ -81687,11 +81687,6 @@ $as_echo "$gxx_include_dir" >&6; }
+ -I$glibcxx_builddir/include \
+ -I$glibcxx_srcdir/libsupc++"
+ 
+-  # For Canadian crosses, pick this up too.
+-  if test $CANADIAN = yes; then
+-    GLIBCXX_INCLUDES="$GLIBCXX_INCLUDES -I\${includedir}"
+-  fi
+-
+   # Stuff in the actual top level.  Currently only used by libsupc++ to
+   # get unwind* headers from the libgcc dir.
+   #TOPLEVEL_INCLUDES='-I$(toplevel_srcdir)/libgcc -I$(toplevel_srcdir)/include'

